### PR TITLE
Fix fs copyFile when source and target are the same

### PR DESCRIPTION
**Pull request by rosen-vladimirov**
When source and target file are the same, copyFile produces empty file.
Add check and if they are the same, do not execute any action.

Move fs tests from appbuilder-cli to common. Add tests for copyFile.

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: test/unit-tests/file-system.ts [hunk 1]
```diff
@@ -0,0 +1,191 @@
+///<reference path="../.d.ts"/>
+"use strict";
+
+import yok = require("../../yok");
```
Comment (teobugslayer) [inline test/unit-tests/file-system.ts]: You may use ES2015 import here

File: test/unit-tests/stubs.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	error(...args: string[]): void {}
+	warn(...args: string[]): void {}
+	info(...args: string[]): void {}
+	debug(...args: string[]): void {}
+	trace(...args: string[]): void {}
+
+	public output = "";
+
+	out(...args: string[]): void {
+		this.output += util.format.apply(null, args) + "\n";
```
Comment (teobugslayer) [inline test/unit-tests/stubs.ts]: Can we use ES6 spread oeprator (...) here?

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by rosen-vladimirov** [comment]
ping @teobugslayer , I've fixed the comments :)

**Comment by teobugslayer** [comment]
:+1:

**Pull request merged**